docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -1,14 +1,574 @@
+
+    ```javascript
+    case 13001:
+        // No one is signed into Office. If the add-in cannot be effectively used when no one 
+        // is logged into Office, then the first call of getAccessToken should pass the 
+        // `allowSignInPrompt: true` option. Since this add-in does that, you should not see
+        // this error. 
+        showMessage("No one is signed into Office. But you can use many of the add-ins functions anyway. If you want to log in, press the Get OneDrive File Names button again.");  
+        break;
+    case 13002:
+        // OfficeRuntime.auth.getAccessToken was called with the allowConsentPrompt 
+        // option set to true. But, the user aborted the consent prompt. 
+        showMessage("You can use many of the add-ins functions even though you have not granted consent. If you want to grant consent, press the Get OneDrive File Names button again."); 
+        break;
+    case 13006:
+        // Only seen in Office on the Web.
+        showMessage("Office on the Web is experiencing a problem. Please sign out of Office, close the browser, and then start again."); 
+        break;
+    case 13008:
+        // The OfficeRuntime.auth.getAccessToken method has already been called and 
+        // that call has not completed yet. Only seen in Office on the Web.
+        showMessage("Office is still working on the last operation. When it completes, try this operation again."); 
+        break;
+    case 13010:
+        // Only seen in Office on the Web.
+        showMessage("Follow the instructions to change your browser's zone configuration.");
+        break;
+    ```
+
+1. Replace `TODO 7` with the following code. or more information about these errors, see [Troubleshoot SSO in Office Add-ins](troubleshoot-sso-in-office-add-ins.md). The function `dialogFallback` invokes the alternative system of authorization. In this add-in, the fallback system opens a dialog which requires the user to sign in, even if the user already is, and uses msal.js and the Implicit Flow to get an access token to Microsoft Graph.
+
+    ```javascript
+    default:
+    // For all other errors, including 13000, 13003, 13005, 13007, 13012, 
+    // and 50001, fall back to non-SSO sign-in.
+    dialogFallback();
+    break;
+    ```
+
+1. Below the `handleClientSideErrors` function, add the following function. 
+
+    ```javascript
+    function handleAADErrors(exchangeResponse) {
+
+    // TODO 8: Handle case where the bootstrap token is expired.
+
+    // TODO 9: Handle all other AAD errors.
+    
+    }
+    ```
+
+1. On rare occasions the bootstrap token that Office has cached is unexpired when Office validates it, but expires by the time it reaches Azure AD for exchange. Azure AD will respond with error **AADSTS500133**. In this case, the add-in should simply recursively recall `getGraphData`. Since the cached bootstrap token is now expired, Office will get a new one from Azure AD. The So replace `TODO 8` with the following. 
```

Comment:
```suggestion
1. On rare occasions the bootstrap token that Office has cached is unexpired when Office validates it, but expires by the time it reaches Azure AD for exchange. Azure AD will respond with error **AADSTS500133**. In this case, the add-in should simply recursively call `getGraphData`. Since the cached bootstrap token is now expired, Office will get a new one from Azure AD. So replace `TODO 8` with the following. 
```